master/buildbot/schedulers/forcesched.py

```diff
@@ -95,6 +95,8 @@ def getFromKwargs(self, kwargs):
         if len(args) == 0:
             if self.required:
                 raise ValidationError("'%s' needs to be specified" % (self.label))
+            if self.readonly:
```

Comment:
Also, I am not sure, that is a correct place for this condition. By name `getFromKwargs` should return something but you've put there a core logic condition.

Let's imagine that, you are a user of this class. When you called `getFromKwargs` you supposed that this method returns some dict or list from kwargs, but when you set read-only property on True, this function nothing return. Is a special corner case which is very ugly. 

Can we move this condition to another place?

Reply:
I can move it to `updateFromKwargs`, this was where I first had it, but it means that we will duplicate the logic a bit for with what is already inside `def getFromKwargs(self, kwargs):` which I did not like.

I'll try to move it back and see if you like that better.